File: 1501~2000/1503. Last Moment Before All Ants Fall Out of a Plank/main.py
"""
Source: https://leetcode.com/problems/last-moment-before-all-ants-fall-out-of-a-plank/
Date: 2023/7/3
Skill:
Ref:
Runtime: 172 ms, faster than 53.10%
Memory Usage: 17.3 MB, less than 94.96%
Time complexity:
Space complexity:
Constraints:

"""

import math
from typing import List, Optional
from collections import defaultdict, Counter, deque
from heapq import heapify, heappush, heappop, nsmallest
import heapq
import functools
from bisect import bisect_left, bisect_right


class Solution:
    def getLastMoment(self, n: int, left: List[int], right: List[int]) -> int:
        left_num = max(left) if len(left) > 0 else 0
        right_num = n - min(right) if len(right) > 0 else 0
        return max(left_num, right_num)


if __name__ == "__main__":
    s = Solution()

File: 1501~2000/1503. Last Moment Before All Ants Fall Out of a Plank/main.cpp
/**
 * Source: t.ly/pLp59
 * Date: 2023/11/4
 * Skill:
 * Runtime: 21 ms, faster than 34.05% of C++ online submissions
 * Memory Usage: 23.04 MB, less than 59.05% of C++ online submissions
 * Time complexity: O(n)
 * Space complexity: O(n)
 * Constraints:
 *
 */


class Solution {
public:
    int getLastMoment(int n, vector<int>& left, vector<int>& right) {
        int res = 0;
        for (auto &i: left) res = max(res, i);
        for (auto &i: right) res = max(res, n - i);
        return res;
    }
};


int main() {
    Solution s;
    vector<vector<int>> arr{{2,0}, {0,2}};
    vector<int> nums{0, 1};
}